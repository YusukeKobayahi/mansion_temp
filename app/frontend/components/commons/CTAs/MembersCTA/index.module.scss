@import "~/styles/variables";
@import "~/styles/utils";

.container {
  margin: 30px calc(50% - 50vw);
  padding: 38px 0 32px;
  background-color: #f8f6f0;
  display: flex;
  justify-content: center;
  align-items: center;

  @media screen and (max-width: $content-max-width) {
    display: block;
  }

  @include screenMd {
    max-width: 100%;
    padding: 30px 16px;
  }
}

.description {
  text-align: center;
  line-height: 1.5;
  margin-bottom: 15px;
  color: #746744;
  font-size: 18px;
  font-weight: bold;

  br {
    &.none {
      display: none;
    }
  }

  .spOnly {
    display: none;
  }

  @include screenMd {
    font-size: 16px;
    margin-bottom: 5px;

    .pcOnly {
      display: none;
    }

    .spOnly {
      display: inline;
    }
  }
}

.btn {
  display: block;
  margin: 0 auto;
  padding: 16px 32px 16px 48px;
  border-radius: 60px;
  font-size: 24px;
  font-weight: bold;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 445px;
  height: 76px;
  color: #ffffff;
  position: relative;
  background-color: #f15e3a;

  @include screenMd {
    width: 100%;
    font-size: 18px;
    padding-left: 50px;
    max-width: 400px;
    padding: 10px 24px 10px 60px;
    height: 65px;
  }

  span {
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: #ffffff;
    color: #f15e3a;
    font-size: 18px;
    width: 68px;
    height: 68px;
    border-radius: 50px;
    position: absolute;
    top: 50%;
    left: 8px;
    transform: translateY(-50%);

    @include screenMd {
      width: 55px;
      height: 55px;
      line-height: 1.3;
      font-size: 16px;
    }
  }

  &::after {
    content: "";
    position: absolute;
    top: 50%;
    right: 30px;
    width: 10px;
    height: 10px;
    border-top: 2px solid #ffffff;
    border-right: 2px solid #ffffff;
    transform: translateY(-50%) rotate(45deg);

    @include screenMd {
      right: 20px;
    }
  }
}

.cta_offer {
  .btn {
    background-color: #ffffff;
    border: 2px solid #f15e3a;
    color: #f15e3a;

    &::after {
      border-color: #f15e3a;
    }

    span {
      background-color: #ffe1d7;
      color: #746744;
    }
  }
}

.cta_register {
  margin-left: 30px;

  .btn {
    background-color: #f15e3a;
  }

  @include screenMd {
    margin-left: 0;
    margin-top: 20px;
  }
}
