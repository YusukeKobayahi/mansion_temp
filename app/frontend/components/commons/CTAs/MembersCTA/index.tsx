import styles from "./index.module.scss";
import classnames from "classnames";
import { clickMembers } from "~/lib/utils";
import { GoogleAnalyticsEvent } from "~/lib/types";

type MembersCTAProps = {
  name: string;
};

const MembersCTA: React.FC<MembersCTAProps> = ({ name }: MembersCTAProps) => {
  const isLongText = (() => {
    if (name.length > 15) return true;
    else return false;
  })();
  const clickCTA = (label: GoogleAnalyticsEvent["label"]) => {
    clickMembers({
      action: "msl_cta_click",
      category: "msl_ichiran",
      label: "msl_lp_left",
      value: 1,
    });
  };
  return (
    <section className={classnames(styles.container, "pcOnly")}>
      <div className={styles.cta_offer}>
        <p className={styles.description}>
          {isLongText && name}＼ {!isLongText && name}の未公開物件が多数 ／
        </p>
        <button className={styles.btn} onClick={() => clickCTA("msl_lp_left")}>
          <span>
            完全
            <br />
            無料
          </span>
          未公開物件の情報が
          <br />
          欲しい方はこちら
        </button>
      </div>
      <div className={styles.cta_register}>
        <p className={styles.description}>
          ＼複数の不動産会社に相談ができる！ ／
        </p>
        <button className={styles.btn} onClick={() => clickCTA("msl_lp_right")}>
          <span>
            完全
            <br />
            無料
          </span>
          会員登録して理想の
          <br />
          マンションを探す
        </button>
      </div>
    </section>
  );
};

export default MembersCTA;
