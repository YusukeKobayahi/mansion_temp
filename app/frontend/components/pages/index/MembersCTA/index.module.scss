@import "~/styles/variables";
@import "~/styles/utils";

.container {
  max-width: 600px;
  margin: 40px auto 0;
  padding: 20px 0;
  display: flex;
  justify-content: center;
  align-items: center;

  @include screenMd {
    width: 100%;
    padding: 30px 16px;
  }
}

.description {
  text-align: center;
  line-height: 1.5;
  margin-bottom: 15px;
  color: #746744;
  font-size: 18px;
  font-weight: bold;

  .spOnly {
    display: none;
    margin-bottom: 5px;
  }

  @include screenMd {
    font-size: 12px;

    .pcOnly {
      display: none;
    }

    .spOnly {
      display: inline;
    }
  }
}

.cta {
  width: 100%;
  text-align: center;
}

.btn {
  display: block;
  margin: 0 auto;
  padding: 16px 32px 16px 48px;
  border-radius: 60px;
  font-size: 24px;
  font-weight: bold;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 445px;
  height: 76px;
  color: #ffffff;
  position: relative;
  background-color: #f15e3a;

  @include screenMd {
    width: 100%;
    font-size: 18px;
    padding-left: 50px;
    max-width: 400px;
    padding: 10px 24px 10px 60px;
    height: 65px;
  }

  span {
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: #ffffff;
    color: #f15e3a;
    font-size: 18px;
    width: 68px;
    height: 68px;
    border-radius: 50px;
    position: absolute;
    top: 50%;
    left: 8px;
    transform: translateY(-50%);
    box-sizing: border-box;

    @include screenMd {
      width: 55px;
      height: 55px;
      line-height: 1.3;
      font-size: 16px;
    }
  }

  &::after {
    content: "";
    position: absolute;
    top: 50%;
    right: 30px;
    width: 10px;
    height: 10px;
    border-top: 2px solid #ffffff;
    border-right: 2px solid #ffffff;
    transform: translateY(-50%) rotate(45deg);

    @include screenMd {
      right: 20px;
    }
  }
}
