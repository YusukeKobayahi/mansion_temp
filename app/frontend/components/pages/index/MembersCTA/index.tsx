import styles from "./index.module.scss";
import { clickMembers } from "~/lib/utils";
import Link from "~/components/commons/Link";

const MembersCTA: React.FC = () => {
  return (
    <section className={styles.container}>
      <div className={styles.cta}>
        <p className={styles.description}>
          ＼ メールアドレス設定と条件入力で会員登録完了 ／
        </p>
        <button
          className={styles.btn}
          onClick={() => {
            clickMembers({
              action: "msl_cta_click",
              category: "msl_top",
              label: "msl_lp",
              value: 1,
            });
          }}
        >
          <span>
            完全
            <br />
            無料
          </span>
          理想の家探しスタート
        </button>
        <Link
          text="Housiiについての詳しい情報はこちらから"
          onClick={() => {
            clickMembers({
              action: "msl_cta_click",
              category: "msl_top",
              label: "msl_lp",
              value: 1,
            });
          }}
        />
      </div>
    </section>
  );
};

export default MembersCTA;
