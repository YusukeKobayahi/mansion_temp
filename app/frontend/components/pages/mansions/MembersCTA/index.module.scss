@import "~/styles/variables";
@import "~/styles/utils";

.container {
  margin: 50px calc(50% - 50vw);
  padding: 38px 0 32px;
  background-color: #f8f6f0;
  justify-content: center;

  @include screenMd {
    margin: 30px 0 0;
    max-width: 100%;
    padding: 30px 16px;
  }
}

.cta_area {
  justify-content: center;
  display: flex;

  @media screen and (max-width: $content-max-width) {
    display: block;
  }
}

.description {
  text-align: center;
  line-height: 1.5;
  margin-bottom: 15px;
  color: #746744;
  font-size: 18px;
  font-weight: bold;

  @include screenMd {
    font-size: 16px;
    margin-bottom: 5px;
  }
}

.btn {
  display: block;
  margin: 0 auto;
  padding: 16px 32px 16px 60px;
  border-radius: 60px;
  font-size: 20px;
  font-weight: bold;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 460px;
  height: 76px;
  color: #ffffff;
  position: relative;
  background-color: #f15e3a;

  @include screenMd {
    width: 100%;
    font-size: 4vw;
    padding-left: 50px;
    max-width: 400px;
    padding: 10px 24px 10px 60px;
    height: 65px;
  }

  span {
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: #ffffff;
    color: #f15e3a;
    font-size: 18px;
    width: 68px;
    height: 68px;
    border-radius: 50px;
    position: absolute;
    top: 50%;
    left: 8px;
    transform: translateY(-50%);

    @include screenMd {
      width: 55px;
      height: 55px;
      line-height: 1.3;
      font-size: 15px;
    }
  }

  &::after {
    content: "";
    position: absolute;
    top: 50%;
    right: 30px;
    width: 10px;
    height: 10px;
    border-top: 2px solid #ffffff;
    border-right: 2px solid #ffffff;
    transform: translateY(-50%) rotate(45deg);

    @include screenMd {
      right: 20px;
    }
  }
}

.cta_offer {
  .btn {
    background-color: #ffffff;
    border: 2px solid #f15e3a;
    color: #f15e3a;

    &::after {
      border-color: #f15e3a;
    }

    span {
      background-color: #ffe1d7;
      color: #746744;
    }
  }
}

.cta_register {
  margin-left: 30px;

  .btn {
    background-color: #f15e3a;
  }

  @include screenMd {
    margin-left: 0;
    margin-top: 20px;
  }
}
