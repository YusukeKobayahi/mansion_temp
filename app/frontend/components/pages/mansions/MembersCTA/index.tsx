import styles from "./index.module.scss";
import { clickMembers } from "~/lib/utils";
import { GoogleAnalyticsEvent } from "~/lib/types";
type MembersCTAProps = {
  mansionName: string;
  pageView: number;
};

const MembersCTA: React.FC<MembersCTAProps> = ({
  mansionName,
  pageView,
}: MembersCTAProps) => {
  const clickCTA = (label: GoogleAnalyticsEvent["label"]) => {
    clickMembers({
      action: "msl_cta_click",
      category: "msl_shosai",
      label: label,
      value: 1,
    });
  };
  return (
    <section className={styles.container}>
      <div>
        <p className={styles.description}>
          {pageView}人がこの物件の購入を検討しています！
        </p>
      </div>
      <div className={styles.cta_area}>
        <div className={styles.cta_offer}>
          <p className={styles.description}>
            {mansionName}をご検討の皆様
            <br />＼ 条件の近い未公開物件があります！ ／
          </p>
          <button
            className={styles.btn}
            onClick={() => {
              clickCTA("msl_lp_souba_left");
            }}
          >
            <span>
              完全
              <br />
              無料
            </span>
            会員登録して物件情報を受け取る
          </button>
        </div>
        <div className={styles.cta_register}>
          <p className={styles.description}>
            複数の不動産会社が
            <br />＼ あなたの理想の物件を提案 ／
          </p>
          <button
            className={styles.btn}
            onClick={() => {
              clickCTA("msl_lp_souba_right");
            }}
          >
            <span>
              完全
              <br />
              無料
            </span>
            会員登録して理想の物件を見つける
          </button>
        </div>
      </div>
    </section>
  );
};

export default MembersCTA;
