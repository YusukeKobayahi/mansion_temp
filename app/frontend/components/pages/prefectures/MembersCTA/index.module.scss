@import "~/styles/variables";
@import "~/styles/utils";

.container {
  max-width: 750px;
  margin: 30px auto;

  @include screenMd {
    max-width: 100%;
    background-color: #f4f2ed;
    padding: 10px;
  }
}

.description {
  text-align: center;
  line-height: 1.5;
  margin-bottom: 15px;
  color: #746744;
  font-size: 18px;
  font-weight: bold;

  .spOnly {
    display: none;
  }

  @include screenMd {
    font-size: 3.2vw;
    margin-bottom: 5px;

    .pcOnly {
      display: none;
    }

    .spOnly {
      display: inline;
    }
  }
}

.btn {
  display: block;
  margin: 0 auto;
  padding: 16px 32px 16px 48px;
  border-radius: 60px;
  font-size: 24px;
  font-weight: bold;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 550px;
  height: 76px;
  color: #ffffff;
  position: relative;
  background-color: #f15e3a;

  @include screenMd {
    width: 100%;
    font-size: 4vw;
    padding-left: 50px;
    max-width: 400px;
    padding: 10px 24px 10px 60px;
    height: 65px;
  }

  span {
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: #ffffff;
    color: #f15e3a;
    font-size: 18px;
    width: 65px;
    height: 65px;
    line-height: 1.2;
    border-radius: 50px;
    position: absolute;
    top: 50%;
    left: 8px;
    transform: translateY(-50%);

    @include screenMd {
      width: 55px;
      height: 55px;
      line-height: 1.3;
      font-size: 16px;
    }
  }

  &::after {
    content: "";
    position: absolute;
    top: 50%;
    right: 30px;
    width: 10px;
    height: 10px;
    border-top: 2px solid #ffffff;
    border-right: 2px solid #ffffff;
    transform: translateY(-50%) rotate(45deg);

    @include screenMd {
      right: 20px;
    }
  }
}
