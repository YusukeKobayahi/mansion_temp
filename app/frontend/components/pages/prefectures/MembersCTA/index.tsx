import styles from "./index.module.scss";
import { clickMembers } from "~/lib/utils";
import { GoogleAnalyticsEvent } from "~/lib/types";

type MembersCTAProps = {
  position: "top" | "bottom";
  subject: string;
};

const MembersCTA: React.FC<MembersCTAProps> = ({
  position,
  subject,
}: MembersCTAProps) => {
  const clickCTA = () => {
    clickMembers({
      action: "msl_cta_click",
      category: "msl_prefectures",
      label: `msl_lp_${position}` as GoogleAnalyticsEvent["label"],
      value: 1,
    });
  };
  return (
    <section className={styles.container}>
      <p className={styles.description}>＼{subject}の未公開物件情報多数／</p>
      <button className={styles.btn} onClick={clickCTA}>
        <span>
          完全
          <br />
          無料
        </span>
        いますぐ会員登録でオファーを待つ
      </button>
    </section>
  );
};

export default MembersCTA;
