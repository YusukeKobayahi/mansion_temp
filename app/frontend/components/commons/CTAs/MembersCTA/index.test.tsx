import { render } from "@testing-library/react";
import MembersCTA from ".";

describe("components/commons/CTAs/MembersCTA", () => {
  it("正常に表示される", () => {
    const dummyData = {
      name: "リブシティ神田",
    };
    const { asFragment } = render(<MembersCTA {...dummyData} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
