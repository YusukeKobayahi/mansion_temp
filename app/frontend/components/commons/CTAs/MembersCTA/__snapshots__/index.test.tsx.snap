// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/commons/CTAs/MembersCTA 正常に表示される 1`] = `
<DocumentFragment>
  <section
    class="container pcOnly"
  >
    <div
      class="cta_offer"
    >
      <p
        class="description"
      >
        ＼ リブシティ神田の未公開物件が多数 ／
      </p>
      <button
        class="btn"
      >
        <span>
          完全
          <br />
          無料
        </span>
        未公開物件の情報が
        <br />
        欲しい方はこちら
      </button>
    </div>
    <div
      class="cta_register"
    >
      <p
        class="description"
      >
        ＼複数の不動産会社に相談ができる！ ／
      </p>
      <button
        class="btn"
      >
        <span>
          完全
          <br />
          無料
        </span>
        会員登録して理想の
        <br />
        マンションを探す
      </button>
    </div>
  </section>
</DocumentFragment>
`;
